web3/providers/eth_tester/main.py

```diff
@@ -135,6 +136,25 @@ def is_connected(self, show_traceback: bool = False) -> Literal[True]:
         return True
 
 
+def _make_response(result: Any, message: str = "") -> RPCResponse:
+    def _error_response(error: str) -> RPCResponse:
+        return RPCResponse(
+            {
+                "id": 1,
+                "jsonrpc": "2.0",
+                "error": RPCError({"code": -32601, "message": error, "data": None}),
```

Comment:
@kclowes I've set data for an error response to None here. I cant think of anything else that we could send back but open to suggestions.

Reply:
I think that makes sense. It's optional, so we could also just skip the `data` key altogether, but either way works. 

Are we using this error response function for errors besides `MethodUnavailable`? I believe the `-32601` error code is reserved for that case only. I'm not sure why I can't link directly to it, but the codes are in section 5.1 of [this doc](https://www.jsonrpc.org/specification) just in case we want to check for that before building the `_error_response`

Reply:
This RPCError response is returned if there is  a `KeyError` or a `NotImplementedError` in `_make_request`.

Reply:
Agreed on the error code here. Maybe we should have a static map of conditions to error codes and apply that based on the exception in some way. Would be good to have the error codes be correct for each case or at least only use this code when the error is due to the method not being available.

Reply:
@fselmo Sounds good to both of your points. As implemented, the endpoint not being defined is treated the same as if the endpoint is there but has not been implemented. As of now, both cases return 32601. Is there a distinction that should be made for each of those cases?

Reply:
Oh, nope, my mistake. Didn't see that `KeyError` in this sense means the method is missing. Looks like the error code is appropriate in both of those cases.

No need to touch any other error code either. If that's requested at some point, we'll cross that bridge. Thanks for the explanation 👌🏼